docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,191 @@
+  2) enclave registration should be feasible only for committed chaincodes
+  end note
 
+  create E_Manager order 15
 
-  ECC1 -> TLCC1 ++: getInfo
-  note over TLCC1
-    This information could also be retrieved elsewhere if easier. At this point in time
-    it does not have to be authenticated (and is anyway passed through untrusted ECC),
-    consistency will be verified later in registration (Remember that TLCC_MR is defined
-    and agreed by human consensus in the ERCC chaincode definition)
+  note over Admin, E_Manager #FFaa00
+  The E_Manager component hides some operations that do not necessarily need to be
+  performed by the Admin -- they could be scripted or embedded in the peer process.
   end note
-  return TLCC_MR, CH_id
 
+  Admin -> E_Manager ++: create CC_Id
+  E_Manager -> ECC1 ++: create
+  'ECC1 -> TLCC1 ++: getInfo
+  'note over TLCC1 #FFaa00
+  '  This information could also be retrieved elsewhere if easier. At this point in time
+  '  it does not have to be authenticated (and is anyway passed through untrusted ECC),
+  '  consistency will be verified later in registration (Remember that TLCC_MR is defined
+  '  and agreed by human consensus in the ERCC chaincode definition)
+  'end note
+  'return TLCC_MR, CH_id
 
-  |||
+  ECC1 -> Enclave1 ++: create
 
-  create participant "ECC_Enclave" as Enclave1 order 30 #99FF99
-  ECC1 -> Enclave1 ++: create (SPID, CC_NAME, CC_VER, Org_1, TLCC_MR, CH_id)
-  note left
-    this is a cgo call
-    Org_1 denotes the peer's MSP_ID
-  end note
+  Enclave1 -> Enclave1: CC_Parameters <- retrieve CC_Id, CC_Version, MSPID, Channel_Id, TLCC_MREnclave
 
-  Enclave1 -> Enclave1 : keyGen
-  note left
-    keyGen produces the following keys:
-    ECC_S_PK : Sig pub key
-    ECC_S_SK : Sig priv key
-    ECC_E_PK : Enc pub key
-    ECC_E_SK : Enc priv key
-  end note
+  'group fpc-cc-tlcc binding (to be discussed)
+  'Enclave1 -> ERCC1 ++: get_tlcc_mrenclave()
+  'return TLCC_MREnclave
+  'end group
 
-  note left Enclave1
-    ECC_ID is //short// unique identifier
-    ECC_ID = Hash(ECC_S_PK)
-  end note
+  Enclave1 -> Enclave1 : generate Private_Keys <- <Enclave_SK, Enclave_DK, CC_DK>
+  Enclave1 -> Enclave1 : Public_Keys <- Public(Private_Keys)
+
+  Enclave1 -> Enclave1 : Credentials <- <CC_Parameters, Public_Keys>
 
-  Enclave1  -> ECC1 ++: binding (ECC_S_PK, ECC_E_PK)
+  group Org-Enclave binding (to be discussed -- not implemented)
+  Enclave1  -> ECC1 ++: binding (ECC_VK)
   ECC1 -> ECC1 : SIG_Org_1 <- create signature
-  note left
+  note left #FFaa00
     Org_1 takes "ownership" of this enclave
-    sign <CC_NAME, CC_VER, Org_1, ECC_S_PK, ECC_E_PK> with Org_1_S_SK
   end note
   return SIG_Org_1
-  |||
+  end group
 
-  Enclave1 -> Enclave1 : generate quote
-  note left
-    quote is defined with reportbody =
-    Hash(<CC_NAME, CC_VER, Org_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, TLCC_MR, CH_id>)
+  Enclave1 -> Enclave1 : Attestation <- attest(Credentials)
```

Comment:
this was already moved into a separate box.
Regarding the signature, it will be part of the certificate.
Also, there seems to be no need to include it in the attestation since the user and/or validation procedure can verify the certificate chain.